scripts/build_upload.py

```diff
@@ -172,7 +172,10 @@ def cdn_upload(local_path, cdn_path, content_type, cdn_token, cdn_id):
     c.setopt(c.HTTPHEADER, ["X-Auth-Token: %s" % cdn_token,
                             "Origin: https://mycloud.rackspace.com",
                             "Content-Type: %s" % content_type])
-    c.setopt(pycurl.POSTFIELDS, open(local_path).read().encode('utf-8'))
+    if binary:
+        c.setopt(pycurl.POSTFIELDS, open(local_path, "rb").read())
+    else:
+        c.setopt(pycurl.POSTFIELDS, open(local_path).read().encode('utf-8'))
```

Comment:
It's a minor thing, but you should try to avoid this kind of duplication. Much better would be:
```
if binary:
    data = open(local_path, "rb").read()
else:
    data = open(local_path).read().encode('utf-8')
c.setopt(pycurl.POSTFIELDS, data)
```